File: blockchain-biz-secure-api/client_api/src/main/java/com/blockchainbiz/app/client_api/model/CertificateDetails.java
package com.blockchainbiz.app.client_api.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class CertificateDetails {
    private String issuer;
    private String validFrom;
    private String validTo;
    private String subject;
    private List<String> subjectAlternativeNames;
    private String hfIban;
}

File: frontend/src/model/CertificateDetails.ts
export interface CertificateDetails{
    issuer: Map<string, string>;
    validFrom: string | null;
    validTo: string | null;
    subject: Map<string, string>;
    subjectAlternativeNames: string[];
    hfIban: Record<string, string>;
}

export interface CertificateDetailsJson{
    issuer: string | null;
    validFrom: string | null;
    validTo: string | null;
    subject: string | null;
    subjectAlternativeNames: string[];
    hfIban: string | null;
}